packages/back-end/src/models/InformationSchemaModel.ts

```diff
@@ -0,0 +1,52 @@
+import mongoose from "mongoose";
+import uniqid from "uniqid";
+import { InformationSchema } from "../types/Integration";
+
+const informationSchema = new mongoose.Schema({
+  id: String,
+  organization: String,
+  databases: [
+    {
+      database_name: String,
+      path: String,
+      schemas: [
+        {
+          schema_name: String,
+          path: String,
+          tables: [
+            {
+              id: String,
+              table_name: String,
+              path: String,
+              columns_id: String,
+            },
+          ],
+        },
+      ],
+    },
+  ],
```

Comment:
If I'm not mistaken, and feel free to correct me if I'm wrong, I don't think schema validation works for nested definitions, which is why I implemented a validator in the models I defined. You can see an example [here](https://github.com/growthbook/growthbook/blob/f92260cac9fa2fbbc857561963d62f33bcc86188/packages/back-end/src/models/EventWebhookModel.ts#L44-L55).

An easy way to test validation is by adding this new model to the [REPL file](https://github.com/growthbook/growthbook/blob/f92260cac9fa2fbbc857561963d62f33bcc86188/packages/back-end/repl.ts) and running `yarn repl` in the backend package and trying to evaluate invalid schema definitions in the console.

Reply:
Updated with the validator.